torch/fx/node.py

```diff
@@ -85,7 +85,7 @@
 
 # TODO: Either refactor this into 2 functions 1 dce for functional graphs and 1 dce for all graphs,
 # or add logic to correctly mark all inplace ops as side effectful.
-_side_effectful_functions: set[Callable] = {
+_side_effectful_functions: set[Callable] = {  # type: ignore[assignment]
     torch._assert,
```

Comment:
Wait, why did this new ignore appear? A bit dodgy, all of these entries appear to be callable?

Reply:
@rec I'm truly confused here; the best I can tell is that `mypy` struggles with auto-detecting `Callable` from mixed sets with different classes of callables, and defaults to `object`. I tried a number of different ideas, and eventually gave up and ignored this one specific place.

Reply:
I gotta try this, this is infuriating. :-D BRB!

Reply:
@rec suggested a way to improve this somewhat, but this represents a bug in `mypy`. I've opened an issue at https://github.com/python/mypy/issues/19091 and will resolve this after committing @rec's suggestion.

Reply:
Does using a 
`_F = TypeVar("_F", bound="Callable[Any, ...]")`
solve the issue?

Reply:
@Skylion007 No, that would be an unbound `TypeVar` in this context. What I've switched to is simply `*_side_effectful_need_to_be_preserved_pre_dispatch`, which apparently works.